cli/commands/completeAllWithdrawals.go

```diff
@@ -0,0 +1,154 @@
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+type TCompleteWithdrawalArgs struct {
+	EthNode     string
+	EigenPod    string
+	Sender      string
+	EstimateGas bool
+}
+
+func DelegationManager(chainId *big.Int) common.Address {
+	data := map[uint64]string{
+		// TODO(zeus) - make this runnable via zeus.
+		1:     "0x39053D51B77DC0d36036Fc1fCc8Cb819df8Ef37A", // mainnet
+		17000: "0x75dfE5B44C2E530568001400D3f704bC8AE350CC", // holesky preprod
+	}
+	contract, ok := data[chainId.Uint64()]
+	if !ok {
+		panic("no delegation manager found for chain")
+	}
+	addr := common.HexToAddress(contract)
+	return addr
+}
+
+func CompleteAllWithdrawalsCommand(args TCompleteWithdrawalArgs) error {
+	ctx := context.Background()
+
+	isSimulation := args.EstimateGas
+
+	eth, err := ethclient.DialContext(ctx, args.EthNode)
+	core.PanicOnError("failed to reach eth node", err)
+
+	chainId, err := eth.ChainID(ctx)
+	core.PanicOnError("failed to load chainId", err)
+
+	acc, err := core.PrepareAccount(&args.Sender, chainId, isSimulation)
+	core.PanicOnError("failed to parse private key", err)
+
+	curBlockNumber, err := eth.BlockNumber(ctx)
+	core.PanicOnError("failed to load current block number", err)
+
+	pod, err := EigenPod.NewEigenPod(common.HexToAddress(args.EigenPod), eth)
+	core.PanicOnError("failed to reach eigenpod", err)
+
+	reg, err := pod.WithdrawableRestakedExecutionLayerGwei(nil)
+	core.PanicOnError("failed to fetch REG", err)
+
+	podOwner, err := pod.PodOwner(nil)
+	core.PanicOnError("failed to read podOwner", err)
+
+	delegationManager, err := IDelegationManager.NewIDelegationManager(DelegationManager(chainId), eth)
+	core.PanicOnError("failed to reach delegation manager", err)
+
+	minDelay, err := delegationManager.MinWithdrawalDelayBlocks(nil)
+	core.PanicOnError("failed to read MinWithdrawalDelayBlocks", err)
+
+	queuedWithdrawals, err := delegationManager.GetQueuedWithdrawals(nil, podOwner)
+	core.PanicOnError("failed to read queuedWithdrawals", err)
+
+	eligibleWithdrawals := lo.Map(queuedWithdrawals.Withdrawals, func(withdrawal IDelegationManager.IDelegationManagerTypesWithdrawal, index int) *IDelegationManager.IDelegationManagerTypesWithdrawal {
+		isBeaconWithdrawal := len(withdrawal.Strategies) == 1 && withdrawal.Strategies[0].Cmp(core.BeaconStrategy()) == 0
+		isExecutable := curBlockNumber >= uint64(withdrawal.StartBlock+minDelay)
+		if isBeaconWithdrawal && isExecutable {
+			return &withdrawal
+		}
+		return nil
+	})
+
+	if len(eligibleWithdrawals) == 0 {
+		fmt.Printf("Your pod has no eligible withdrawals.\n")
+		return nil
+	}
+
+	var runningSum uint64 = 0
+	affordedWithdrawals := lo.Map(eligibleWithdrawals, func(withdrawal *IDelegationManager.IDelegationManagerTypesWithdrawal, index int) *IDelegationManager.IDelegationManagerTypesWithdrawal {
+		if withdrawal == nil {
+			return nil
+		}
+		withdrawalShares := queuedWithdrawals.Shares[index][0].Uint64()
+		if reg < (runningSum + withdrawalShares) {
```

Comment:
this is wrong. withdrawalShares is in wei and reg is gwei

Reply:
good catch